HardwareObjects/abstract/AbstractActuator.py

```
@@ -143,7 +142,7 @@ def update_limits(self, limits=None):
         Args:
             limits (tuple): two floats tuple (low limit, high limit).
         """
-        if limits is None:
+        if not limits:
```

Actually bot are not correct. We should check against a tuple of (None,None).

---

Good point - I think you found a bug, but maybe not just the one you thought ;-) ?

Thinking about it, if the limits passed in are None, we clearly need to call get_limits(). 
If the limits passed in are (None, None), OTOH, I think we should set them. (None,None) means 'do not use limits', it is the default value, and it must be possible to set the value back to 'do not use limits'. 

My 'if limits' only make a difference if the limits passed in are 0 or False or () (empty tuple).
0 and False are arguably bugs, and should raise an error. If someone passes in an empty tuple the code would wok in either case, but I think it is probalbly better to call get_limits, than to silently do nothing.

How do you think this ought to look?